src/elf/parser.rs

```diff
@@ -532,10 +576,48 @@ impl ElfParser {
     }
 
     /// Retrieve the data corresponding to the ELF section at index `idx`.
+    #[inline]
     pub fn section_data(&self, idx: usize) -> Result<&[u8]> {
         self.cache.section_data(idx)
     }
 
+    /// Retrieve the data corresponding to the ELF section at index
+    /// `idx`, optionally decompressing it if it is compressed.
+    ///
+    /// If the section is compressed the resulting decompressed data
+    /// will be cached for the life time of this object.
+    pub fn section_data_decompressed(&self, idx: usize) -> Result<&[u8]> {
```

Comment:
is there a real need to keep section_data and section_data_decompressed separately? Why not teach section_data to transparently decompress (if corresponding feature was selected, of course), and keep things simple?

Reply:
No strong reason. Will change it.